microprofile.jdt/org.eclipse.lsp4mp.jdt.core/src/main/java/org/eclipse/lsp4mp/jdt/internal/config/java/MicroProfileConfigDiagnosticsParticipant.java

```diff
@@ -0,0 +1,267 @@
+							e);
+				}
+			}
+			return false;
+		}
+
+		/**
+		 * Validate "defaultValue" attribute of <code>@ConfigProperty</code> and
+		 * generate diagnostics if "defaultValue" cannot be represented by the given
+		 * field type.
+		 *
+		 * See
+		 * https://github.com/eclipse/microprofile-config/blob/master/spec/src/main/asciidoc/converters.asciidoc
+		 * for more details on default converters.
+		 *
+		 * @param annotation       the annotation to validate the defaultValue of
+		 * @param defaultValueExpr the default value expression, or null if no default
+		 *                         value is defined
+		 */
+		private void validatePropertyDefaultValue(Annotation annotation, Expression defaultValueExpr) {
+			try {
+				FieldDeclaration fieldDeclaration = (FieldDeclaration) annotation.getParent();
+				if (defaultValueExpr != null) {
+					String defValue = ((StringLiteral) defaultValueExpr).getLiteralValue();
+					ITypeBinding fieldBinding = fieldDeclaration.getType().resolveBinding();
+					if (fieldBinding != null && !isAssignable(fieldBinding, defValue)) {
+						Range range = context.getUtils().toRange(context.getTypeRoot(),
+								defaultValueExpr.getStartPosition(), defaultValueExpr.getLength());
+						Diagnostic d = new Diagnostic(
+								range, "'" + defValue + "'" + " does not match the expected type of '"
+										+ fieldBinding.getName() + "'.",
+								DiagnosticSeverity.Error, MICRO_PROFILE_CONFIG_DIAGNOSTIC_SOURCE);
+						d.setCode(MicroProfileConfigErrorCode.DEFAULT_VALUE_IS_WRONG_TYPE.getCode());
+						diagnostics.add(d);
+					}
+				}
+			} catch (JavaModelException e) {
+				LOGGER.log(Level.WARNING,
+						"Exception while calculating diagnostics for @ConfigProperty (default value has correct type)",
+						e);
+			}
+		}
+
+		/**
+		 * Generates diagnostics if the property that this <code>@ConfigProperty</code>
+		 * defines does not have a value assigned to it.
+		 *
+		 * @param annotation       the ConfigProperty annotation
+		 * @param defaultValueExpr the default value expression, or null if no default
+		 *                         value is defined
+		 */
+		private void validatePropertyHasValue(Annotation annotation, Expression defaultValueExpr) {
+			try {
+				String name = null;
+				Expression nameExpression = getAnnotationMemberValueExpression(annotation,
+						CONFIG_PROPERTY_ANNOTATION_NAME);
+				boolean hasDefaultValue = defaultValueExpr != null;
+
+				if (nameExpression != null) {
+					name = ((StringLiteral) nameExpression).getLiteralValue();
+				}
+
+				if (name != null && !hasDefaultValue && !doesPropertyHaveValue(name, context)
+						&& !isPropertyIgnored(name)) {
+					String fileUri = context.getUri();
+					IOpenable openable = context.getUtils().resolveCompilationUnit(fileUri);
+					Range range = context.getUtils().toRange(openable, annotation.getStartPosition(),
```

Comment:
It should be nice to use the name parameter as range instead of Configproperty annotation range (like validatePropertyDefaultValue does with Expression)